lib/exercises.js

```diff
@@ -1,33 +1,81 @@
 // This method will return an array of arrays.
 // Each subarray will have strings which are anagrams of each other
-// Time Complexity: ?
-// Space Complexity: ?
-function grouped_anagrams(strings) {
-  throw new Error("Method hasn't been implemented yet!");
+// Time Complexity: O(n) where n is the length of the array passed in as the argument "strings" because each item in the array is visited once
+// Space Complexity: O(1) (***Chris, is this right? It takes 2 times the space of 'strings' which reduces to 1. Or is it O(n) because the space taken is is 2 x n where n is the length of the 'strings' array?***)
```

Comment:
I would say O(n) because you are building a JavaScript object which scales based on the size of the array.